batch/batch/server/database.py

```diff
@@ -159,6 +159,23 @@ class BatchTable(Table):
     async def has_record(self, id):
         return await super().has_record({'id': id})
 
+    async def find_records(self, complete=None, success=None):
+        values = (,)
+        sql = "select * from {self.name} as batch"
+        if complete or success:
+            sql += " inner join {self._db.batch_jobs.name} as bj using (batch_id)"
+            sql += " inner join {self._db.jobs.name} as job using (job_id)"
+            if complete:
+                values += "Complete"
+                sql += " where job.state = %s"
```

Comment:
Don't you want the opposite if complete is false? I don't think you return disjoint subsets.